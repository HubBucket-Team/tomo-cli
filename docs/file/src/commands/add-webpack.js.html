<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../">
  <title data-ice="title">src/commands/add-webpack.js | tomo-cli</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
<meta name="description" content="A friendly command line tool designed to help create sustainable software using web technology"><meta property="twitter:card" content="summary"><meta property="twitter:title" content="tomo-cli"><meta property="twitter:description" content="A friendly command line tool designed to help create sustainable software using web technology"></head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./">Home</a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  <a href="test.html" data-ice="testLink">Test</a>
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
<a style="position:relative; top:3px;" href="https://github.com/jhwohlgemuth/tomo-cli"><img width="20px" src="./image/github.png"></a></header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/ui.js~UI.html">UI</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-CommandError">CommandError</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Debug">Debug</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-OfflineWarning">OfflineWarning</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Task">Task</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-TaskList">TaskList</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-Warning">Warning</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-populateQueue">populateQueue</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#commands">commands</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addA11y">addA11y</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addBabel">addBabel</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addBrowsersync">addBrowsersync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addEsdoc">addEsdoc</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addJest">addJest</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addMakefile">addMakefile</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addPostcss">addPostcss</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-removePostcss">removePostcss</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-rustTasks">rustTasks</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-addWebpack">addWebpack</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-removeWebpack">removeWebpack</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-typedef">T</span><span data-ice="name"><span><a href="typedef/index.html#static-typedef-task">task</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#commands-add-eslint">commands/add-eslint</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-tasks">tasks</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#commands-add-marionette">commands/add-marionette</a><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-tasks">tasks</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/BasicEditor.js~BasicEditor.html">BasicEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/MakefileEditor.js~MakefileEditor.html">MakefileEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-class">C</span><span data-ice="name"><span><a href="class/src/utils/Scaffolder.js~Scaffolder.html">Scaffolder</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoExist">allDoExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoExistSync">allDoExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoNotExist">allDoNotExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-allDoNotExistSync">allDoNotExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-format">format</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getBinDirectory">getBinDirectory</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-parse">parse</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-someDoExist">someDoExist</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-someDoExistSync">someDoExistSync</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-createJsonEditor">createJsonEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-createModuleEditor">createModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getIntendedInput">getIntendedInput</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getVersions">getVersions</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-install">install</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-uninstall">uninstall</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-verifyRustInstallation">verifyRustInstallation</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-BabelConfigModuleEditor">BabelConfigModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-EslintConfigModuleEditor">EslintConfigModuleEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-PackageJsonEditor">PackageJsonEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-PostcssConfigEditor">PostcssConfigEditor</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-variable">V</span><span data-ice="name"><span><a href="variable/index.html#static-variable-WebpackConfigEditor">WebpackConfigEditor</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/commands/add-webpack.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import {
    PackageJsonEditor,
    WebpackConfigEditor,
    install,
    uninstall
} from &apos;../utils&apos;;
import {allDoExist, allDoExistSync, allDoNotExist, someDoExist} from &apos;../utils/common&apos;;

const WEBPACK_DEPENDENCIES = [
    &apos;cpy-cli&apos;,
    &apos;del-cli&apos;,
    &apos;npm-run-all&apos;,
    &apos;webpack&apos;,
    &apos;webpack-cli&apos;,
    &apos;webpack-dashboard&apos;,
    &apos;webpack-jarvis&apos;,
    &apos;webpack-dev-server&apos;,
    &apos;babel-loader&apos;
];
/**
 * @type {task[]}
 * @see https://webpack.js.org/
 */
export const addWebpack = [
    {
        text: &apos;Create Webpack configuration file&apos;,
        task: async ({sourceDirectory}) =&gt; {
            const entry = {
                app: `&apos;${sourceDirectory}/main.js&apos;`
            };
            const resolve = {
                modules: `[resolve(__dirname, &apos;${sourceDirectory}&apos;), &apos;node_modules&apos;]`
            };
            await (new WebpackConfigEditor())
                .create()
                .prepend(`const DashboardPlugin = require(&apos;webpack-dashboard/plugin&apos;);`)
                .prepend(`const {resolve} = require(&apos;path&apos;);`)
                .prepend(`/* eslint-env node */`)
                .extend({entry, resolve})
                .commit();
        },
        condition: () =&gt; allDoNotExist(&apos;webpack.config.js&apos;)
    },
    {
        text: &apos;Add build tasks to package.json&apos;,
        task: async ({outputDirectory, sourceDirectory}) =&gt; {
            const scripts = {
                copy: &apos;npm-run-all --parallel copy:assets copy:index&apos;,
                &apos;copy:assets&apos;: `cpy &apos;./assets/!(css)/**/*.*&apos; &apos;./assets/**/[.]*&apos; ${outputDirectory} --parents --recursive`,
                &apos;copy:index&apos;: `cpy &apos;./assets/index.html&apos; ${outputDirectory}`,
                prebuild: `del-cli ${outputDirectory}/assets`,
                build: &apos;webpack&apos;,
                postbuild: &apos;npm run copy&apos;,
                &apos;build:watch&apos;: `watch &apos;npm run build&apos; ${sourceDirectory}`,
                &apos;build:dashboard&apos;: &apos;webpack-dashboard -- webpack-dev-server --config ./webpack.config.js&apos;
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();

        },
        condition: () =&gt; someDoExist(&apos;package.json&apos;)
    },
    {
        text: &apos;Add dev task to package.json&apos;,
        task: async () =&gt; {
            const scripts = {
                dev: &apos;npm-run-all --parallel build:watch build:css:watch&apos;
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () =&gt; allDoExist(&apos;package.json&apos;, &apos;postcss.config.js&apos;),
        optional: () =&gt; allDoExistSync(&apos;package.json&apos;, &apos;postcss.config.js&apos;)
    },
    {
        text: &apos;Install Webpack dependencies&apos;,
        task: ({skipInstall}) =&gt; install(WEBPACK_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({isNotOffline}) =&gt; isNotOffline &amp;&amp; someDoExist(&apos;package.json&apos;)
    }
];
export const removeWebpack = [
    {
        text: &apos;Delete Webpack configuration file&apos;,
        task: async () =&gt; {
            await (new WebpackConfigEditor())
                .delete()
                .commit();
        },
        condition: () =&gt; someDoExist(&apos;webpack.config.js&apos;)
    },
    {
        text: &apos;Remove Webpack build tasks from package.json&apos;,
        task: async () =&gt; {
            const scripts = {
                copy: undefined,
                &apos;copy:assets&apos;: undefined,
                &apos;copy:index&apos;: undefined,
                prebuild: undefined,
                build: undefined,
                postbuild: undefined,
                &apos;build:watch&apos;: undefined,
                &apos;build:dashboard&apos;: undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();

        },
        condition: () =&gt; someDoExist(&apos;package.json&apos;)
    },
    {
        text: &apos;Uninstall Webpack dependencies&apos;,
        task: () =&gt; uninstall(WEBPACK_DEPENDENCIES),
        condition: ({skipInstall}) =&gt; !skipInstall &amp;&amp; someDoExist(&apos;package.json&apos;) &amp;&amp; (new PackageJsonEditor()).hasAll(...WEBPACK_DEPENDENCIES),
        optional: ({skipInstall}) =&gt; !skipInstall
    }
];
export default addWebpack;</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
