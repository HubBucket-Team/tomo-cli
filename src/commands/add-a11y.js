import {
    PackageJsonEditor,
    install,
    uninstall
} from '../utils';
import {allDoExist} from '../utils/common';
/**
 * @type {task[]}
 * @see http://pa11y.org/
 */
export const addA11y = [
    {
        text: 'Add pa11y tasks to package.json',
        task: async ({outputDirectory}) => {
            const scripts = {
                'lint:a11y': `pa11y ${outputDirectory}/index.html`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Install pa11y',
        task: ({skipInstall}) => install(['pa11y'], {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && allDoExist('package.json')
    }
];
export const removeA11y = [
    {
        text: 'Remove pa11y tasks from package.json',
        task: async () => {
            const scripts = {
                'lint:a11y': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall pa11y',
        task: () => uninstall(['pa11y']),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll('pa11y'),
        optional: ({skipInstall}) => !skipInstall
    }
];
export default addA11y;