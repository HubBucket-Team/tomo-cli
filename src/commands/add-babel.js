import {
    allDoNotExist,
    BabelConfigModuleEditor,
    install,
    someDoExist
} from '../utils';

const cfg = new BabelConfigModuleEditor();

const BABEL_DEPENDENCIES = [
    '@babel/cli',
    '@babel/core',
    '@babel/runtime'
];
const BABEL_PRESETS = [
    '@babel/preset-env'
];
const BABEL_PLUGINS = [
    '@babel/plugin-transform-runtime',
    '@babel/plugin-proposal-class-properties',
    '@babel/plugin-proposal-export-default-from',
    '@babel/plugin-proposal-optional-chaining'
];
const BABEL_REACT_PRESET = [
    '@babel/preset-react'
];
export default [
    {
        text: 'Create Babel config file',
        task: () => cfg.create(),
        condition: () => allDoNotExist('babel.config.js', '.babelrc', '.babelrc.js')
    },
    {
        text: 'Install Babel dependencies',
        task: ({skipInstall}) => install(BABEL_DEPENDENCIES, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel presets',
        task: ({skipInstall}) => install(BABEL_PRESETS, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel plugins',
        task: ({skipInstall}) => install(BABEL_PLUGINS, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel React preset',
        task: ({skipInstall}) => install(BABEL_REACT_PRESET, {dev: true, skipInstall}),
        condition: ({useReact}) => (useReact && someDoExist('package.json')),
        optional: ({useReact}) => useReact
    },
    {
        text: 'Add React support to Babel configuration file',
        task: () => cfg.extend({
            presets: [...BABEL_PRESETS, ...BABEL_REACT_PRESET]
        }),
        condition: ({useReact}) => (useReact && someDoExist('babel.config.js')),
        optional: ({useReact}) => useReact
    }
];