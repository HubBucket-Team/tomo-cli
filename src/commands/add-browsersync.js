import {PackageJsonEditor, install} from '../utils';
import {allDoExist, someDoExist} from '../utils/common';

const BROWSERSYNC_DEPENDENCIES = [
    'browser-sync',
    'npm-run-all'
];
/**
 * @type {task[]}
 * @see https://www.browsersync.io/docs/command-line
 */
export const addBrowsersync = [
    {
        text: 'Add Browsersync tasks to package.json',
        task: async ({outputDirectory}) => {
            const scripts = {
                prestart: 'npm run build',
                start: 'npm-run-all --parallel build:watch build:css:watch serve:dev',
                'serve:dev': `browser-sync start --server ${outputDirectory} --files ${outputDirectory}`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json', 'webpack.config.js', 'postcss.config.js')
    },
    {
        text: 'Install Browsersync dependencies',
        task: ({skipInstall}) => install(BROWSERSYNC_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && someDoExist('package.json')
    }
];