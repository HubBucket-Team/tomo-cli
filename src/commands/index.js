/* eslint-disable no-magic-numbers */
import {
    createPackageJson,
    createSourceDirectory
} from './common';
import addA11y from './add-a11y';
import addBabel from './add-babel';
import addEsdoc from './add-esdoc';
import addEslint from './add-eslint';
import addJest from './add-jest';
import addMarionette from './add-marionette';
import addPostcss from './add-postcss';
import addWebpack from './add-webpack';

const createProject = [
    ...createPackageJson,
    ...createSourceDirectory,
    ...addBabel,
    ...addEslint,
    ...addJest
];
const create = {
    project: createProject,
    app: [
        ...createProject,
        ...addMarionette
    ],
    server: [
        ...createProject
    ]
};
const add = {
    a11y: addA11y,
    babel: addBabel,
    docs: addEsdoc,
    eslint: [
        ...addBabel,
        ...addEslint
    ],
    jest: [
        ...addBabel,
        ...addJest
    ],
    postcss: addPostcss,
    webpack: addWebpack
};

module.exports = {
    add,
    create,
    new: create// alias for create
};