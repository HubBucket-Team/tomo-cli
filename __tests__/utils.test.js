import {
    EslintConfigModuleEditor,
    getIntendedInput,
    getVersions,
    install,
    PackageJsonEditor
} from '../src/utils';
import {join} from 'path';
import crypto from 'crypto';
import {mkdirp} from 'fs-extra';
import rimraf from 'rimraf';
import {tmpdir} from 'os';
import execa from 'execa';
import commands from '../src/commands';

jest.mock('execa');

const testDirectory = join(__dirname, 'fixtures');

describe('package.json mem-fs editor', () => {
    let pkg;
    beforeEach(() => {
        pkg = new PackageJsonEditor(testDirectory);
    });
    test('create', () => {
        pkg = new PackageJsonEditor('/some/directory');
        expect(pkg.read()).toEqual('');
        expect(pkg.create(false).read()).toMatchSnapshot();
    });
    test('read', () => {
        expect(pkg.read()).toMatchSnapshot();
    });
    test('extend', () => {
        expect(pkg.extend({scripts: {foo: 'bar'}}, false).read()).toMatchSnapshot();
    });
    test('copy', () => {
        const newDirPath = join(testDirectory, 'new');
        pkg.copy(newDirPath, false);
        expect(pkg.fs.readJSON(join(newDirPath, 'package.json'))).toMatchSnapshot();
    });
    test('delete', () => {
        pkg = new PackageJsonEditor('/some/directory');
        expect(pkg.read()).toEqual('');
        expect(pkg.create(false).read()).toMatchSnapshot();
        expect(pkg.delete(false).read()).toEqual('');
    });
});
describe('.eslintrc.js mem-fs editor', () => {
    let cfg;
    beforeEach(() => {
        cfg = new EslintConfigModuleEditor(testDirectory);
    });
    test('create', () => {
        cfg = new EslintConfigModuleEditor('/some/directory');
        expect(cfg.read()).toEqual('');
        expect(cfg.create(false).read()).toMatchSnapshot();
    });
    test('read', () => {
        expect(cfg.read()).toMatchSnapshot();
    });
    test('extend', () => {
        expect(cfg.read()).toMatchSnapshot();
        expect(cfg.extend({key: {foo: `'foo'`}}, false).read()).toMatchSnapshot();
        expect(cfg.extend({key: {bar: `'bar'`}}, false).read()).toMatchSnapshot();
    });
    test('delete', () => {
        cfg = new EslintConfigModuleEditor('/some/directory');
        expect(cfg.read()).toEqual('');
        expect(cfg.create(false).read()).toMatchSnapshot();
        expect(cfg.delete(false).read()).toEqual('');
    });
    test('prepend', () => {
        expect(cfg.prepend(`const {existsSync} = require('fs-extra');`, false).read()).toMatchSnapshot();
        expect(cfg.prepend(`const {join} = require('path');`, false).read()).toMatchSnapshot();
        expect(cfg.extend({key: {baz: 'baz'}}, false).read()).toMatchSnapshot();
    });
});
describe('File & folder scaffolder', () => {
    let tempDirectory;
    beforeEach(async () => {
        tempDirectory = join(tmpdir(), `tomo-test-${crypto.randomBytes(20).toString('hex')}`);// eslint-disable-line no-magic-numbers
        await mkdirp(tempDirectory);
        process.chdir(tempDirectory);
    });
    afterEach(done => {
        rimraf(tempDirectory, done);
    });
    test('can copy files', async () => {
    });
});
describe('getIntendedInput', () => {
    test('return closest matches', () => {
        const [intendedCommand, intendedTerms] = getIntendedInput(commands, 'ad', ['lint']);
        expect(intendedCommand).toEqual('add');
        expect(intendedTerms).toEqual(['eslint']);
    });
});
describe('getVersions', () => {
    test('handle no module name', async () => {
        expect(await getVersions()).toEqual([]);
    });
    test('format response from npm', async () => {
        const stdout = '1.0.0 ,\n 2.0.0 ,\n 3.0.0 ';
        execa.mockResolvedValue({stdout});
        expect(await getVersions('some-module-name')).toMatchSnapshot();
    });
});
describe('install', () => {
    test('handle array of string names', async () => {
        const latest = true;
        const skipInstall = true;
        const dev = true;
        expect(await install()).toEqual(['install']);
        expect(await install(['some-module'])).toEqual(['install', 'some-module@latest']);
        expect(await install(['some-module'])).toEqual(['install', 'some-module@latest']);
        expect(await install(['some-module'], {latest: false, skipInstall})).toEqual(['install', 'some-module']);
        expect(await install(['foo', 'bar'], {latest: false, skipInstall})).toEqual(['install', 'foo', 'bar']);
        expect(await install(['some-module'], {dev, skipInstall})).toEqual(['install', 'some-module', '--save-dev']);
        expect(await install(['some-module'], {dev, latest, skipInstall})).toEqual(['install', 'some-module@latest', '--save-dev']);
        expect(await install(['foo', 'bar'], {latest: false, dev, skipInstall})).toEqual(['install', 'foo', 'bar', '--save-dev']);
    });
});
