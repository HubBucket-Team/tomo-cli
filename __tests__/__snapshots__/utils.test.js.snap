// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.eslintrc.js mem-fs editor create 1`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint'
};"
`;

exports[`.eslintrc.js mem-fs editor delete 1`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint',
    key: {
        foo: 'foo',
        bar: 'bar'
    }
};"
`;

exports[`.eslintrc.js mem-fs editor extend 1`] = `
"const {join} = require('path');
const {existsSync} = require('fs-extra');

module.exports = {
    env: {es6: true, jest: true},
    extends: ['omaha-prime-grade'],
    parser: 'babel-eslint',
    key: {foo: 'foo', bar: 'bar', baz: baz}
};"
`;

exports[`.eslintrc.js mem-fs editor extend 2`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint',
    key: {
        foo: 'foo'
    }
};"
`;

exports[`.eslintrc.js mem-fs editor extend 3`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint',
    key: {
        foo: 'foo',
        bar: 'bar'
    }
};"
`;

exports[`.eslintrc.js mem-fs editor prepend 1`] = `
"const {existsSync} = require('fs-extra');

module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint',
    key: {
        foo: 'foo',
        bar: 'bar'
    }
};"
`;

exports[`.eslintrc.js mem-fs editor prepend 2`] = `
"const {join} = require('path');
const {existsSync} = require('fs-extra');

module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint',
    key: {
        foo: 'foo',
        bar: 'bar'
    }
};"
`;

exports[`.eslintrc.js mem-fs editor prepend 3`] = `
"const {join} = require('path');
const {existsSync} = require('fs-extra');

module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint',
    key: {
        foo: 'foo',
        bar: 'bar',
        baz: baz
    }
};"
`;

exports[`.eslintrc.js mem-fs editor read 1`] = `
"const {join} = require('path');
const {existsSync} = require('fs-extra');

module.exports = {
    env: {es6: true, jest: true},
    extends: ['omaha-prime-grade'],
    parser: 'babel-eslint',
    key: {foo: 'foo', bar: 'bar', baz: baz}
};"
`;

exports[`Makefile editor addComment 1`] = `
"# Makefile created with tomo
# Knowledge of the Holy One is understanding"
`;

exports[`Makefile editor addTask 1`] = `
"# Makefile created with tomo
foo:
	echo foo
bar:
	echo bar"
`;

exports[`Makefile editor append 1`] = `
"# Makefile created with tomo
test line"
`;

exports[`Makefile editor appendScripts (no bin variable) 1`] = `
"# Makefile created with tomo

test:
	@jest --coverage

lint:
	@eslint . -c ./.eslintrc.js

lint-watch:
	@echo Starting lint:watch
	@watch 'npm run lint' src

build:
	@echo Starting build
	@webpack
	@echo Build complete
"
`;

exports[`Makefile editor appendScripts (with bin variable) 1`] = `
"# Makefile created with tomo
bin := home/user/project/__tests__/fixtures/node_modules/.bin/

test:
	@jest --coverage

lint:
	@$(bin)eslint . -c ./.eslintrc.js

lint-watch:
	@echo Starting lint:watch
	@watch 'npm run lint' src

build:
	@echo Starting build
	@$(bin)webpack
	@echo Build complete
"
`;

exports[`Makefile editor create 1`] = `"# Makefile created with tomo"`;

exports[`Makefile editor importScripts 1`] = `Object {}`;

exports[`Makefile editor importScripts 2`] = `
Object {
  "build": "webpack",
  "lint": "eslint . -c ./.eslintrc.js",
  "lint:watch": "watch 'npm run lint' src",
  "postbuild": "echo Build complete",
  "prebuild": "echo Starting build",
  "prelint:watch": "echo Starting lint:watch",
  "test": "jest --coverage",
}
`;

exports[`getVersions format response from npm (only allow valid version strings) 1`] = `
Array [
  "1.0.0",
  "2.0.0",
  "3.0.0",
]
`;

exports[`package.json mem-fs editor copy 1`] = `
Object {
  "author": "",
  "dependencies": Object {
    "chalk": "2.4.2",
    "delay": "4.2.0",
    "execa": "1.0.0",
  },
  "description": "",
  "devDependencies": Object {
    "@babel/cli": "^7.4.3",
    "@babel/core": "^7.4.3",
    "eslint": "^5.16.0",
    "webpack": "4.30.0",
  },
  "keywords": Array [],
  "license": "ISC",
  "main": "index.js",
  "name": "fixtures",
  "scripts": Object {
    "build": "webpack",
    "lint": "eslint . -c ./.eslintrc.js",
    "lint:watch": "watch 'npm run lint' src",
    "postbuild": "echo Build complete",
    "prebuild": "echo Starting build",
    "prelint:watch": "echo Starting lint:watch",
    "test": "jest --coverage",
  },
  "version": "1.0.0",
}
`;

exports[`package.json mem-fs editor create 1`] = `
Object {
  "description": "A super cool app/server/tool/library/widget/thingy",
  "keywords": Array [],
  "license": "MIT",
  "name": "my-project",
  "version": "0.0.0",
}
`;

exports[`package.json mem-fs editor delete 1`] = `
Object {
  "description": "A super cool app/server/tool/library/widget/thingy",
  "keywords": Array [],
  "license": "MIT",
  "name": "my-project",
  "version": "0.0.0",
}
`;

exports[`package.json mem-fs editor extend 1`] = `
Object {
  "author": "",
  "dependencies": Object {
    "chalk": "2.4.2",
    "delay": "4.2.0",
    "execa": "1.0.0",
  },
  "description": "",
  "devDependencies": Object {
    "@babel/cli": "^7.4.3",
    "@babel/core": "^7.4.3",
    "eslint": "^5.16.0",
    "webpack": "4.30.0",
  },
  "keywords": Array [],
  "license": "ISC",
  "main": "index.js",
  "name": "fixtures",
  "scripts": Object {
    "build": "webpack",
    "foo": "bar",
    "lint": "eslint . -c ./.eslintrc.js",
    "lint:watch": "watch 'npm run lint' src",
    "postbuild": "echo Build complete",
    "prebuild": "echo Starting build",
    "prelint:watch": "echo Starting lint:watch",
    "test": "jest --coverage",
  },
  "version": "1.0.0",
}
`;

exports[`package.json mem-fs editor read 1`] = `
Object {
  "author": "",
  "dependencies": Object {
    "chalk": "2.4.2",
    "delay": "4.2.0",
    "execa": "1.0.0",
  },
  "description": "",
  "devDependencies": Object {
    "@babel/cli": "^7.4.3",
    "@babel/core": "^7.4.3",
    "eslint": "^5.16.0",
    "webpack": "4.30.0",
  },
  "keywords": Array [],
  "license": "ISC",
  "main": "index.js",
  "name": "fixtures",
  "scripts": Object {
    "build": "webpack",
    "lint": "eslint . -c ./.eslintrc.js",
    "lint:watch": "watch 'npm run lint' src",
    "postbuild": "echo Build complete",
    "prebuild": "echo Starting build",
    "prelint:watch": "echo Starting lint:watch",
    "test": "jest --coverage",
  },
  "version": "1.0.0",
}
`;
